airbyte-integrations/connectors/source-open-exchange-rates/source_open_exchange_rates/source.py

```diff
@@ -0,0 +1,119 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import pendulum
+from pendulum import DateTime
+
+import requests
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+
+
+class OpenExchangeRates(HttpStream, ABC):
+    url_base = "https://openexchangerates.org/api/"
+
+    primary_key = None
+    date_field_name = "timestamp"
+    cursor_field = date_field_name
+
+
+    def __init__(self, base: Optional[str], start_date: str, app_id: str) -> None:
+        super().__init__()
+
+        self.base = base
+        self.start_date = start_date
+        self.app_id = app_id
+        self._cursor_value = None
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+
+        params = {}
+
+        if self.base is not None:
+            params["base"] = self.base
+
+        return params
+
+    def request_headers(self, **kwargs) -> MutableMapping[str, Any]:
+        headers = {"Authorization": f"Token {self.app_id}", "Content-Type": "application/json"}
+
+        return headers
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        response_json = response.json()
+        yield response_json
+
+    def stream_slices(self, stream_state: Mapping[str, Any] = None, **kwargs) -> Iterable[Optional[Mapping[str, Any]]]:
+        stream_state = stream_state or {}
+        start_date = pendulum.parse(stream_state.get(self.date_field_name, self.start_date))
+        return self._chunk_date_range(start_date)
+
+    def path(
+        self,
+        stream_state: Mapping[str, Any] = None,
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return f"historical/{stream_slice[self.date_field_name]}.json"
+
+
+    def _chunk_date_range(self, start_date: DateTime) -> List[Mapping[str, Any]]:
+        """
+        Returns a list of each day between the start date and now.
+        The return value is a list of dicts {'date': date_string}.
+        """
+        dates = []
+        while start_date < pendulum.now():
+            dates.append({"timestamp": start_date.to_date_string()})
```

Comment:
minor naming nitpick, but this is not really a timestamp but rather a date, it would probably be cleared to rename it to  `date`